pii_recognition/experiments/pii_validation/google_cloud_reports/mistakes_ner.json

```diff
@@ -0,0 +1,3580 @@
+    "address": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "48145": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "3": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "Valadouro 3, Ubide 48145": {
+      "type": "LOCATION",
+      "score": 0.21,
+      "src": "ground_truth"
+    }
+  },
+  "I want to cancel my card 4556733991563087 because I lost it": {
+    "card": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "4556733991563087": {
+      "type": "CREDIT_CARD",
+      "score": 0.0,
+      "src": "ground_truth"
+    }
+  },
+  "One of the most depressing songs on the list. He's injured from the waist down from Mozambique, but Malvin just has to get laid. Don't go to town, Bojka!": {
+    "One": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "list": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "waist": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "town": { "type": "LOCATION", "score": 0.0, "src": "predicted" },
+    "Bojka": { "type": "PERSON", "score": 0.0, "src": "ground_truth" }
+  },
+  "Can I withdraw cash using my card 4532272104540970 at aTM center ?": {
+    "cash": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "card": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "center": { "type": "LOCATION", "score": 0.0, "src": "predicted" },
+    "4532272104540970": {
+      "type": "CREDIT_CARD",
+      "score": 0.0,
+      "src": "ground_truth"
+    }
+  },
+  "how do i change my address to p.o. box 194, upernavik 3962 for post mail?": {
+    "address": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "p.o.": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "box": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "post mail": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "194": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "3962": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "p.o. box 194, upernavik 3962": {
+      "type": "LOCATION",
+      "score": 0.0,
+      "src": "ground_truth"
+    }
+  },
+  "It's like that since 12/17/1967": {
+    "12": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "17": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "1967": { "type": "NUMBER", "score": 0.0, "src": "predicted" },
+    "12/17/1967": { "type": "BIRTHDAY", "score": 0.2, "src": "ground_truth" }
+  },
+  "C'mon, sing it with me: \"You picked a fine time to leave me Paige, four hungry children and a crop in the field...\"": {
+    "crop": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "field": { "type": "OTHER", "score": 0.0, "src": "predicted" },
+    "children": { "type": "PERSON", "score": 0.0, "src": "predicted" },
+    "four": { "type": "NUMBER", "score": 0.0, "src": "predicted" }
+  },
+  "You can tell Romina was a huge Romina Ruelas fan. Written when he was only 14.": {
```

Comment:
![image](https://user-images.githubusercontent.com/2088399/99197807-c3743180-27e8-11eb-8db0-3667d6d72a36.png)

Reply:
This is the raw protobuf results I got (before passing to any parsers)
```
entities {
  name: "Romina"
  type_: PERSON
  salience: 0.8519809246063232
  mentions {
    text {
      content: "Romina"
      begin_offset: 13
    }
    type_: PROPER
  }
  mentions {
    text {
      content: "fan"
      begin_offset: 45
    }
    type_: COMMON
  }
}
entities {
  name: "Romina Ruelas"
  type_: PERSON
  salience: 0.14801909029483795
  mentions {
    text {
      content: "Romina Ruelas"
      begin_offset: 31
    }
    type_: PROPER
  }
}
entities {
  name: "14"
  type_: NUMBER
  metadata {
    key: "value"
    value: "14"
  }
  mentions {
    text {
      content: "14"
      begin_offset: 75
    }
  }
}
language: "en"
```

`Romina Ruelas` and `Romina` (precision is 1.0) are "perfect" predictions, so they are not logged into the mistakes. However, based on the data structure, `fan` seems to be a PERSON entity but it does not make any sense to me.

Reply:
I think we might need to do some fancier processing of the google results to score them properly. The fact that `fan` is a`COMMON` reference to `Romina` is correct, we just don't care. Potentially we should ignore the `COMMON` type references.

Reply:
You are right. I've removed `COMMON` and kept `PROPER` and `TYPE_UNKNOWN`, the performance of Google NL model has improved a lot but still not as good as `Comprehend`. Documentation of `Choice of PII Model` has updated to reflect the changes, so does the mistakes file in this PR.